compiler-core/src/language_server/tests/compilation.rs

```diff
@@ -1,95 +1,157 @@
-use super::*;
+use super::common::*;
 
 #[test]
-fn compile_please() {
+fn compile_sends_correct_actions() {
     let io = LanguageServerTestIO::new();
-    let mut engine = setup_engine(&io);
+    let mut engine = setup_language_server(&io);
+
+    let _response = engine.compile_please();
+    drop(engine);
+
+    let actions_during_new = vec![
+        Action::DependencyDownloadingStarted,
+        Action::DownloadDependencies,
+        Action::DependencyDownloadingFinished,
+        Action::LockBuild,
+        Action::UnlockBuild,
+    ];
+    let actions_during_compile = vec![
+        Action::CompilationStarted,
+        Action::LockBuild,
+        Action::UnlockBuild,
+        Action::CompilationFinished,
+    ];
+
+    let actual_actions = io.into_actions();
+    assert_eq!(
+        actual_actions,
+        [actions_during_new, actions_during_compile].concat()
+    );
+}
+
+#[test]
+fn compile_project_successfully() {
+    let io = LanguageServerTestIO::new();
+    let mut engine = setup_language_server(&io);
```

Comment:
This method creates an engine instance rather than a language server instance, let's revert to the previous name.